processor/tailsamplingprocessor/internal/sampling/ottl_test.go

```diff
@@ -0,0 +1,122 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package sampling
+
+import (
+	"context"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"go.opentelemetry.io/collector/component/componenttest"
+	"go.opentelemetry.io/collector/pdata/pcommon"
+	"go.opentelemetry.io/collector/pdata/ptrace"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/pkg/ottl"
+)
+
+func TestEvaluate_OTTL(t *testing.T) {
+	traceID := pcommon.TraceID([16]byte{1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16})
+
+	cases := []struct {
+		Desc                string
+		SpanConditions      []string
+		SpanEventConditions []string
+		Spans               []spanWithAttributes
+		Decision            Decision
+	}{
+		{
+			// policy
+			"OTTL conditions not set",
+			[]string{},
+			[]string{},
+			[]spanWithAttributes{{SpanAttributes: map[string]string{"attr_k_1": "attr_v_1"}}},
+			NotSampled,
+		},
+		{
+			"OTTL conditions match specific span attributes 1",
+			[]string{"attributes[\"attr_k_1\"] == \"attr_v_1\""},
+			[]string{},
+			[]spanWithAttributes{{SpanAttributes: map[string]string{"attr_k_1": "attr_v_1"}}},
+			Sampled,
+		},
+		{
+			"OTTL conditions match specific span attributes 2",
```

Comment:
I'm missing a case where `attributes["attr_k_1"] != "attr_v_1"` and the span attribute value would be `attr_v_2". This would yield `Sampled`, right?

Reply:
That's correct. New case added as well.